tiledb/sm/filesystem/s3.h

```diff
@@ -451,11 +451,31 @@ class S3 {
         , key(std::move(in_key))
         , upload_id(std::move(in_upload_id))
         , completed_parts(std::move(in_completed_parts))
-        , st(Status::Ok()) {
+        , st(Status::Ok())
+        , mtx(){};
```

Comment:
e don't need to default construct objects that have default constructors and we follow the pattern of omitting these in the list initializer elsewhere in the code. Remove for consistency.